doc/releases/changelog-dev.md

```diff
@@ -13,6 +13,19 @@
 
 <h3>Improvements</h3>
 
+* `DiagonalOperations` is not a separate subclass any more. 
+  [(#1889)](https://github.com/PennyLaneAI/pennylane/pull/1889) 
```

Comment:
I suppose this should go into breaking changes?

Reply:
True!

Reply:
Anything part of the official developer API should be considered a breaking change :)

Since developers are users too